crates/db_views_actor/src/community_view.rs

```diff
@@ -231,11 +232,16 @@ impl<'a> CommunityQueryBuilder<'a> {
           .then_order_by(community_aggregates::published.desc())
       }
     };
-
-    if !self.show_nsfw.unwrap_or(false) {
+    if !self.show_nsfw.unwrap_or(false) && person_id_join == PersonId(-1) {
```

Comment:
There's an `if self.my_person_id.is_some() {` below, these should go in sub-blocks there. Don't do the same check twice if unavoidable.

Reply:
Done